ccm_web/server/src/api/api.interfaces.ts

```diff
@@ -34,14 +33,13 @@ export function isAPIErrorData (value: unknown): value is APIErrorData {
 }
 
 export interface ExternalUserData {
-  [email: string]: {
-    userCreated: CanvasUser | APIErrorData | false
-    invited?: CirrusInvitationResponse | CirrusErrorData
-  }
+  email: string
+  userCreated: boolean | APIErrorPayload
+  invited?: true | CirrusErrorData
 }
 
 interface ExternalUserCreationResultBase {
-  data: ExternalUserData
+  data: ExternalUserData[]
```

Comment:
I changed this to 1) remove user data not necessary for the client and 2) use an array instead of an object. Explanation for 1: We use an admin token for these actions, so it made sense to minimize users' access to this data if it wasn't being used. Explanation for 2: I realized when working with the response data that it would be nice to have the objects be self-describing, and that arrays would enable simpler syntax when iterating through the data.

Reply:
Not exactly sure why I thought the keys would be better. Sorry for the go-around on that.